pkg/mtbroker/filter/filter_handler.go

```diff
@@ -283,28 +192,74 @@ func (r *Handler) sendEvent(ctx context.Context, tctx cloudevents.HTTPTransportC
 	if extErr := event.ExtensionAs(broker.EventArrivalTime, &arrivalTimeStr); extErr == nil {
 		arrivalTime, err := time.Parse(time.RFC3339, arrivalTimeStr)
 		if err == nil {
-			r.reporter.ReportEventProcessingTime(reportArgs, time.Since(arrivalTime))
+			_ = h.reporter.ReportEventProcessingTime(reportArgs, time.Since(arrivalTime))
 		}
 	}
 
-	sendingCTX := utils.SendingContextFrom(ctx, tctx, subscriberURI.URL())
+	// Send the event to the subscriber
+	target := subscriberURI.String()
+	req, err := h.sender.NewCloudEventRequestWithTarget(ctx, target)
+	if err != nil {
+		h.logger.Info("failed to create the request", zap.String("target", target))
+		writer.WriteHeader(http.StatusInternalServerError)
+		return
+	}
+	req.Header = utils.PassThroughHeaders(request.Header)
+	err = cehttp.WriteRequest(ctx, binding.ToMessage(event), req)
+	if err != nil {
+		h.logger.Info("failed to write request")
+		writer.WriteHeader(http.StatusInternalServerError)
+		return
+	}
 
 	start := time.Now()
-	rctx, replyEvent, err := r.ceClient.Send(sendingCTX, *event)
-	rtctx := cloudevents.HTTPTransportContextFrom(rctx)
-	// Record the dispatch time.
-	r.reporter.ReportEventDispatchTime(reportArgs, rtctx.StatusCode, time.Since(start))
-	// Record the event count.
-	r.reporter.ReportEventCount(reportArgs, rtctx.StatusCode)
-	// Wrap any errors along with the response status code so that can be propagated upstream.
+	resp, err := h.sender.Send(req)
```

Comment:
You can wrap this logic that counts the elapsed time while sending the event in another method

Reply:
Together with the headers pass through and even conversion i think

Reply:
Not sure, try it and see what best fits, but this code is a little bit too much spaghetti